websites/H/HentaiVN/presence.ts

```diff
@@ -0,0 +1,499 @@
+const presence = new Presence({
+		clientId: "942782020055089192"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+let page = "1",
+	keyword,
+	namekey,
+	doukey,
+	charkey,
+	keystr,
+	largebox;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+		startTimestamp: browsingTimestamp
+	};
+
+	if (new URLSearchParams(window.location.search).has("key")) {
+		presenceData.details = "Đang tìm kiếm:";
+		presenceData.state = document.querySelector("input").value;
+		presenceData.smallImageKey = "search";
+	} else if (document.location.pathname === "/") {
+		if (document.location.search.includes("page"))
+			page = new URLSearchParams(document.location.search).get("page");
+
+		presenceData.details = "Đang duyệt trang chủ";
+		presenceData.state = `Trang ${page}`;
+		presenceData.smallImageKey = "surf";
+	} else if (document.location.pathname.startsWith("/tacgia=")) {
+		if (document.location.search.includes("page"))
+			page = new URLSearchParams(document.location.search).get("page");
+
+		presenceData.details = `Đang xem danh sách truyện của tác giả ${document
+			.querySelectorAll("span[itemprop='name']")[2]
+			.textContent.trim()}`;
+		presenceData.state = `Trang ${page}`;
+		presenceData.smallImageKey = "list";
+	} else if (document.location.pathname.startsWith("/char=")) {
+		if (document.location.search.includes("page"))
+			page = new URLSearchParams(document.location.search).get("page");
+
+		presenceData.details = `Đang xem danh sách truyện có nhân vật ${document
+			.querySelectorAll("span[itemprop='name']")[2]
+			.textContent.trim()}`;
+		presenceData.state = `Trang ${page}`;
+		presenceData.smallImageKey = "list";
+	} else if (document.location.pathname.startsWith("/doujin=")) {
+		if (document.location.search.includes("page"))
+			page = new URLSearchParams(document.location.search).get("page");
+
+		presenceData.details = `Đang xem danh sách truyện là doujin ${document
```

Comment:
```suggestion
		presenceData.details = `Đang xem danh sách doujin ${document
```
Conciseness

Reply:
That is the term used in the website, which cannot be changed.